eth/filters/api.go

```diff
@@ -412,6 +413,9 @@ func (api *FilterAPI) GetFilterChanges(id rpc.ID) (interface{}, error) {
 	api.filtersMu.Lock()
 	defer api.filtersMu.Unlock()
 
+	chainConfig := api.sys.backend.ChainConfig()
+	latest := api.sys.backend.CurrentHeader()
```

Comment:
This will be problematic. There might have been multiple blocks produced since the last time you retrieved the changes, in which case `NewRPCPendingTransaction` will use the latest head for calculating the basefee and whatnot fields vs the one that was actually present when those past blocks were produced.

The only correct place to initialize these fields would be when they are added to the list. Even then it's incorrect to pull the latest block at that point, rather it should be sent along with the transaction event.

Reply:
I noticed this, but I found that #26126 has ignored it. To elaborate, the basefee solely impacts mined transactions, we solely rely on the latest block number to derive the sender.

With this in mind, I suggest that we also ignore this problem here, and fix them together at a later time.